nat/service_ics.go

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package nat
+
+import (
+	"net"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+
+	log "github.com/cihub/seelog"
+	"github.com/pkg/errors"
+)
+
+const (
+	enablePublicSharing  = "$config.EnableSharing(0)"
+	enablePrivateSharing = "$config.EnableSharing(1)"
+	disableSharing       = "$config.DisableSharing()"
+)
+
+type serviceICS struct {
+	mu                 sync.Mutex
+	ifaces             map[string]RuleForwarding // list in internal interfaces with enabled internet connection sharing
+	remoteAccessStatus string
+	powerShell         func(cmd string) ([]byte, error)
+}
+
+// Enable enables internet connection sharing for the public interface.
+func (nat *serviceICS) Enable() error {
```

Comment:
Would be so nice to have single-purpose functions like `enableRemoteAccess()` and `Enable()` - which just glues everything together